typescript/infra/src/core/deploy.ts

```diff
@@ -2,177 +2,205 @@ import { ethers } from 'ethers';
 import path from 'path';
 
 import {
-  AbacusConnectionManager,
   AbacusConnectionManager__factory,
-  Inbox,
   InboxValidatorManager,
   InboxValidatorManager__factory,
   Inbox__factory,
   InterchainGasPaymaster__factory,
-  OutboxValidatorManager,
   OutboxValidatorManager__factory,
   Outbox__factory,
-  UpgradeBeaconController,
   UpgradeBeaconController__factory,
 } from '@abacus-network/core';
 import { AbacusAppDeployer, ProxiedContract } from '@abacus-network/deploy';
 import {
   AbacusCore,
+  ChainMap,
   ChainName,
   CoreContractAddresses,
-  ProxiedAddress,
+  CoreContracts,
+  DomainConnection,
+  InboxContracts,
+  MailboxAddresses,
+  MultiProvider,
+  RemoteChainMap,
+  Remotes,
+  domains,
+  utils as sdkUtils,
 } from '@abacus-network/sdk';
 import { types } from '@abacus-network/utils';
 
 import { DeployEnvironment, RustConfig } from '../config';
 
-import { CoreConfig, ValidatorManagerConfig } from './types';
+import { ValidatorManagerConfig } from './types';
 
-export class AbacusCoreDeployer extends AbacusAppDeployer<
-  CoreContractAddresses,
-  CoreConfig
+export class AbacusCoreDeployer<
```

Comment:
This is a separate ticket, but would it make sense to have `hardhat`'s TestRouterDeploy to be basically `AbacusCoreDeployer` just with different contracts? Obviously out of scope for this PR, but maybe worth a ticket

Reply:
https://github.com/abacus-network/abacus-monorepo/issues/383